tdrs-frontend/src/components/FileUpload/FileUpload.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { useDispatch, useSelector } from 'react-redux'
+import { clearError, upload } from '../../actions/reports'
+
+function FileUpload({ section }) {
+  // e.g. 'Aggregate Case Data' => 'aggregate-case-data'
+  // The set of uploaded files in our Redux state
+  const files = useSelector((state) => state.reports.files)
+  const dispatch = useDispatch()
+
+  // e.g. "1 - Active Case Data" => ["1", "Active Case Data"]
+  const [sectionNumber, sectionName] = section.split(' - ')
+
+  const file = files.find((currentFile) => sectionName === currentFile.section)
+
+  const formattedSectionName = file.section
+    .split(' ')
+    .map((word) => word.toLowerCase())
+    .join('-')
+
+  const fileName = file?.fileName
+  const hasUploadedFile = Boolean(fileName)
+
+  const ariaDescription = hasUploadedFile
+    ? `Selected File ${file?.fileName}. To change the selected file, click this button.`
+    : `Drag file here or choose from folder.`
+
+  const uploadFile = ({ target }) => {
+    dispatch(clearError({ section: target.name }))
+    dispatch(
+      upload({
+        file: target.files[0],
+        section: target.name,
+      })
+    )
+  }
+
+  return (
+    <div
+      className={`usa-form-group ${file.error ? 'usa-form-group--error' : ''}`}
+    >
+      <label className="usa-label text-bold" htmlFor={formattedSectionName}>
+        Section {sectionNumber} - {sectionName}
+      </label>
+      <div>
+        {file.error && (
+          <div
+            className="usa-error-message"
+            id={`${formattedSectionName}-error-alert`}
+            role="alert"
+          >
+            {file.error.message}
+          </div>
+        )}
+      </div>
+      <div
+        id={`${formattedSectionName}-file`}
+        aria-hidden
```

Comment:
Do we want the `ariaDescription` to have this `aria-hidden` class? 

Looks like `aria-hidden` will make the content unavailable, "removes that element and all of its children from the accessibility tree."

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-hidden_attribute

Reply:
Yes this is deliberate so it is read for only screen readers.

Reply:
https://www.accessibility-developer-guide.com/examples/sensible-aria-usage/describedby/

The `aria-describedby` attribute in the `input` below references this hidden div so that it is read by screen readers while hidden normally.

Reply:
Ah, thanks for the explanation! 😄